File: examples/go/hello2-ksm.go
package main

import (
	"os"
	"time"

	ksm "github.com/keeper-security/secrets-manager-go/core"
)

func main() {
	// One time tokens can be used only once - afterwards use the generated config.json
	hostname := "keepersecurity.com"
	token := "<ONE TIME TOKEN>"
	sm := ksm.NewSecretsManagerFromSettings(token, hostname, true)
	// sm := ksm.NewSecretsManagerFromConfig(ksm.NewFileKeyValueStorage("client-config.json"))

	allRecords, err := sm.GetSecrets([]string{})
	if err != nil {
		println("Error retrieving all records: ", err.Error())
		os.Exit(1)
	}

	// Get first record
	recToUpdate := allRecords[0]

	// Get first field in a record of type password
	passwordField := map[string]interface{}{}
	if passwordFields := recToUpdate.GetFieldsByType("password"); len(passwordFields) > 0 {
		passwordField = passwordFields[0]
	}

	if len(passwordField) > 0 {
		newPassword := "New Password from hello world - " + time.Now().Format(time.RFC850)
		recToUpdate.SetPassword(newPassword)

		updatedRawJson := ksm.DictToJson(recToUpdate.RecordDict)
		recToUpdate.RawJson = updatedRawJson

		// Perform save operation
		if err := sm.Save(recToUpdate); err != nil {
			println("Error saving record: ", err.Error())
		}
	} else {
		println("No password field found in selected record")
	}

	println("Get only one record by UID")
	if foundRecord, err := sm.GetSecrets([]string{"<RECORD UID>"}); err == nil {
		if len(foundRecord) > 0 {
			println(foundRecord[0].RawJson)
		} else {
			println("Record doesn't exist.")
		}
	} else {
		println("Error retrieveing single record: ", err.Error())
	}
}

File: examples/python/hello2-ksm.py
from datetime import datetime

from keeper_secrets_manager_core import SecretsManager
from keeper_secrets_manager_core.storage import FileKeyValueStorage, InMemoryKeyValueStorage
from keeper_secrets_manager_core.utils import json_to_dict, dict_to_json

if __name__ == '__main__':

    secrets_manager = SecretsManager(
## if your Keeper Account is in other region than US, update the hostname accordingly
        hostname='keepersecurity.com',
        token='<ONE TIME TOKEN>',
        config=FileKeyValueStorage('config2.json')
    )

    all_records = secrets_manager.get_secrets()

    # Get first record
    rec_to_update = all_records[0]

    raw_json = rec_to_update.raw_json
    raw_dict = json_to_dict(raw_json)

    fields = raw_dict['fields']

    # getting first field in a record of type password
    password_field = next((item for item in fields if item["type"] == "password"), None)

    if password_field:
        password_field['value'] = ["New Password from hello world -" + str(datetime.now())]

        updated_raw_json = dict_to_json(raw_dict)

        rec_to_update.raw_json = updated_raw_json

        # Perform save operation
        secrets_manager.save(rec_to_update)
    else:
        print("No records w/ password field was found")

    print("Get only one record by UID")

    found_record = secrets_manager.get_secrets(['<RECORD UID>'])

    print(found_record)
